src/traces/box/set_positions.js

```diff
@@ -86,23 +86,26 @@ function setPositionOffset(traceType, gd, boxList, posAxis, pad) {
     // check for forced minimum dtick
     Axes.minDtick(posAxis, boxdv.minDiff, boxdv.vals[0], true);
 
-    // set the width of all boxes
-    for(i = 0; i < boxList.length; i++) {
-        calcTrace = calcdata[boxList[i]];
-        calcTrace[0].t.dPos = dPos;
-    }
-
     var gap = fullLayout[traceType + 'gap'];
     var groupgap = fullLayout[traceType + 'groupgap'];
     var padfactor = (1 - gap) * (1 - groupgap) * dPos / fullLayout[numKey];
 
     // autoscale the x axis - including space for points if they're off the side
     // TODO: this will overdo it if the outermost boxes don't have
     // their points as far out as the other boxes
-    Axes.expand(posAxis, boxdv.vals, {
+    var extremes = Axes.findExtremes(posAxis, boxdv.vals, {
         vpadminus: dPos + pad[0] * padfactor,
         vpadplus: dPos + pad[1] * padfactor
     });
+
+    for(i = 0; i < boxList.length; i++) {
+        calcTrace = calcdata[boxList[i]];
+        // set the width of all boxes
+        calcTrace[0].t.dPos = dPos;
+        // link extremes to all boxes
+        calcTrace[0].trace._extremes[posAxis._id] = extremes;
```

Comment:
same question as for bars... can we just put these extremes in the first one?

Reply:
... for _consistency_ I'm leaning on voting for making all traces have an `_extremes` container. Like in my last comment, I can't imagine `getAutorange` ever being a performance drag.